app/uk/gov/hmrc/servicedependencies/persistence/derived/DerivedDependencyRepository.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.servicedependencies.persistence.derived
+
+import com.google.inject.Inject
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.{Filters, IndexModel, IndexOptions, Indexes, ReplaceOneModel, ReplaceOptions}
+import org.mongodb.scala.model.Filters.{equal, nin, or}
+import org.mongodb.scala.model.Indexes.compoundIndex
+import play.api.Logging
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.servicedependencies.model.RepoType.Service
+import uk.gov.hmrc.servicedependencies.model.{ApiServiceDependencyFormats, DependencyScope, MetaArtefactDependency, RepoType, SlugInfoFlag}
+import uk.gov.hmrc.servicedependencies.persistence.DeploymentRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DerivedDependencyRepository @Inject()(
+                                      mongoComponent: MongoComponent,
+                                      deploymentRepository: DeploymentRepository
+                                    )(implicit ec: ExecutionContext
+                                    ) extends PlayMongoRepository[MetaArtefactDependency](
+  collectionName = "DERIVED-dependencies"
+  , mongoComponent = mongoComponent
+  , domainFormat   = MetaArtefactDependency.mongoFormat
+  , indexes        = Seq(
+    IndexModel(Indexes.ascending("slugName"), IndexOptions().name("slugNameIdx")),
+    IndexModel(Indexes.ascending("group"),    IndexOptions().name("groupIdx")),
+    IndexModel(Indexes.ascending("artefact"), IndexOptions().name("artefactIdx")),
+    IndexModel(Indexes.ascending("repoType"), IndexOptions().name("repoIdx")),
+  ) ++ DependencyScope.values.map(s => IndexModel(Indexes.ascending(s.asString + "Flag"), IndexOptions().name(s.asString + "FlagIdx")))
+) with Logging {
+
+  // automatically refreshed when given new meta data artefacts from update scheduler
+  override lazy val requiresTtlIndex = false
+
+  def addAndReplace(dependencies: Seq[MetaArtefactDependency]): Future[Unit] = {
+    collection
+        .bulkWrite(
+          dependencies.map(d =>
+            ReplaceOneModel(
+              filter = Filters.and(
+                equal("slugName",     d.slugName),
```

Comment:
Given this repository is to handle the non-slug cases, we should probably call this `repoName`